moto/iam/models.py

```diff
@@ -2120,6 +2122,29 @@ def delete_group(self, group_name):
                 "The group with name {0} cannot be found.".format(group_name)
             )
 
+    def update_group(self, group_name, new_group_name, new_path="/"):
+        if new_group_name in self.groups:
+            raise IAMConflictException(
+                "Group {0} already exists".format(new_group_name)
+            )
+        try:
+            group = self.groups[group_name]
+        except KeyError:
+            raise IAMNotFoundException(
+                "The group with name {0} cannot be found.".format(group_name)
+            )
+
+        existing_policies = group.managed_policies.copy()
+        for policy_arn in existing_policies:
+            self.detach_group_policy(policy_arn, group_name)
+        if new_path:
+            group.path = new_path
+        if new_group_name:
+            group.name = new_group_name
+            self.groups[new_group_name] = self.groups.pop(group_name)
+        for policy_arn in existing_policies:
+            self.attach_group_policy(policy_arn, new_group_name)
```

Comment:
`new_group_name` could be None, so this call could fail. 

From what I can see the whole re-attaching business is only necessary when updating the name, so this can be rewritten as:
```
if new_group_name:
    existing_policies = group.managed_policies.copy()
    # detach policies...
    # update name...
    # attach policies...
```

Reply:
Thanks @bblommers.. Good point. Actually the whole logic is only necessary only if `new_group_name` exists..